README.md

```diff
@@ -131,15 +141,364 @@ class Admin < Dill::Role
 end
 ```
 
-##To Be Continued...
+That's it for the walkthrough, but Dill has a lot more to offer. We recommend digging into Widgets next, as Widget is the fundamental class common to all the Dill elements.
+
+
+#Widgets<a name="widgets"></a>
+A widget is the fundamental Dill element. A widget wraps a DOM element, allowing you to call methods on that element, like checking for text or clicking.
+
+
+## Widget Declaration
+A widget can be declared with the `widget` macro, or as a class. We prefer the macro declaration where possible.
+
+Maco widget declarations take a CSS selector as the first argument.
+
+```ruby
+  widget :todo_item, '.todo-item'
+```
+
+Class widget declarations define the widget with CSS selectors via the `root` macro.
+
+```ruby
+class TodoItem < Dill::Widget
+  root '.todo-item'
+end
+```
+
+## Nested Widgets
+Widgets can be defined within widgets. CSS classes of child widgets are scoped within the class of the parent widget.
+
+Child widgets are available via the `widget` method.
+
+```ruby
+class TodoManager < Dill::Role
+  widget :todo_item, '.todo-item' do
+    widget :delete_link, '.delete' # :delete_link looks for a .delete class within a .todo-item element
+  end
+
+  # to access the delete_link widget:
+  # widget(:todo_item).widget(:delete_link)
```

Comment:
Interesting. This is a common pattern these days on Teespring. Would you recommend always wrapping access in a method? Or is there a better way?